src/server/scripts/Outland/BlackTemple/boss_shade_of_akama.cpp

```diff
@@ -157,236 +210,151 @@ class boss_shade_of_akama : public CreatureScript
-                events.ScheduleEvent(EVENT_RESET_ENCOUNTER, 5000);
-            }
-
-            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
-
-            me->SetWalk(true);
             Initialize();
+            events.ScheduleEvent(EVENT_INITIALIZE_SPAWNERS, Seconds(1));
+            me->SummonCreatureGroup(SUMMON_GROUP_RESET);
         }
 
-        void EnterCombat(Unit* /*who*/) override { }
-
-        void AttackStart(Unit* who) override
+        void EnterEvadeMode(EvadeReason /*why*/) override
         {
-            if (me->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE))
-            {
-                if (Creature* Akama = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_AKAMA_SHADE)))
-                    if (Akama->IsAlive())
-                        ScriptedAI::AttackStart(Akama);
-            }
-            else
-                ScriptedAI::AttackStart(who);
+            _Reset();
+
+            for (GuidVector::const_iterator itr = Spawners.begin(); itr != Spawners.end(); ++itr)
+                if (Creature* Spawner = ObjectAccessor::GetCreature(*me, *itr))
+                    Spawner->AI()->DoAction(ACTION_DESPAWN_ALL_SPAWNS);
+
+            _DespawnAtEvade();
         }
 
         void DoAction(int32 actionId) override
         {
-            if (actionId == ACTION_CHANNELER_DIED)
-                me->RemoveAuraFromStack(SPELL_SHADE_SOUL_CHANNEL_2);
-
-            UpdateSpeed();
+            if (actionId == ACTION_AKAMA_DEAD)
+                EnterEvadeMode(EVADE_REASON_OTHER);
         }
 
         void SpellHit(Unit* /*caster*/, SpellInfo const* spell) override
         {
             if (spell->Id == SPELL_AKAMA_SOUL_CHANNEL)
             {
-                combatStarted = true;
-                events.ScheduleEvent(EVENT_START_ATTACK_AKAMA, 500);
-                events.ScheduleEvent(EVENT_SET_CHANNELERS_SPAWNERS, 1000);
+                events.ScheduleEvent(EVENT_START_CHANNELERS_AND_SPAWNERS, Seconds(1));
                 me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_STATE_NONE);
+                events.ScheduleEvent(EVENT_EVADE_CHECK, Seconds(10));
                 if (Creature* Akama = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_AKAMA_SHADE)))
-                    me->AddThreat(Akama, 10000000.0f);
+                {
+                    AttackStart(Akama);
+                    me->AddThreat(Akama, 100000.0f);
+                }
             }
-            else if (spell->Id == SPELL_SHADE_SOUL_CHANNEL_2)
-                UpdateSpeed();
+
+            if (spell->Id == SPELL_AKAMA_SOUL_EXPEL)
+                DoCastSelf(SPELL_AKAMA_SOUL_EXPEL_CHANNEL);
         }
 
-        void UpdateSpeed()
+        void MoveInLineOfSight(Unit* who) override
         {
-            float moveSpeed = 0.2f;
-
-            if (me->GetAuraCount(SPELL_SHADE_SOUL_CHANNEL_2) <= 3)
+            if (who->GetGUID() == instance->GetGuidData(DATA_AKAMA_SHADE) && me->IsWithinMeleeRange(who))
```

Comment:
There should be a `MovementInform` for the movechase when he gets to melee range.